web/modules/custom/dbcdk_community/src/Plugin/Block/FlaggedContentList.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/**
+ * @file
+ * FlaggedContentList class definition.
+ */
+
+namespace Drupal\dbcdk_community\Plugin\Block;
+
+use DBCDK\CommunityServices\Api\FlagApi;
+use DBCDK\CommunityServices\ApiClient;
+use Drupal\Core\Block\BlockBase;
+use Drupal\dbcdk_community\Content\FlaggableContent;
+use Drupal\dbcdk_community\Content\FlaggableContentRepository;
+
+/**
+ * Provides a block displaying a list of flagged content.
+ *
+ * @Block(
+ *   id = "flagged_content_list",
+ *   admin_label = @Translation("Flagged content list"),
+ * )
+ */
+class FlaggedContentList extends BlockBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    /* var FlaggableContentRepository $flagged_content */
+    $flagged_content = \Drupal::service('dbcdk_community.content.flaggable_content_repository');
+    $list = $flagged_content->getContentWithUnreadFlags();
+
+    $content_per_page = 10;
+    $page = pager_default_initialize(count($list), $content_per_page);
+
+    $table = [
+      '#theme' => 'table',
+      '#caption' => $this->t(
+        'Showing %from-%to of %total pieces of flagged content', [
+          '%from' => ($page * $content_per_page) + 1,
+          '%to' => ($page + 1) * $content_per_page,
+          '%total' => count($list),
+        ]),
+      '#header' => [
+        $this->t('Content'),
+        $this->t('Active flags'),
+        $this->t('Latest flag'),
+        $this->t('Flag comment')
+      ],
+      '#data' => [],
+      '#empty' => $this->t('There is currently no flagged content'),
+    ];
+
+    /* @var FlaggableContent[] $page_content */
+    $page_content = array_slice($list, $page, $content_per_page);
+    foreach ($page_content as $c) {
```

Comment:
Could we make `$c` more descriptive by naming it "$row" or similar?
By doing this then you know that it's ment to be a row in the ['#rows'] array